pkg/mongoproxy/plugins/schema/util.go

```diff
@@ -141,6 +144,31 @@ func Mapify(d bson.D) bson.M {
 	return m
 }
 
+// Map creates a map from the elements of the D with operator
+// It makes additional process for arrays
+func MapifyWithOp(d bson.D, m bson.M) bson.M {
+	for _, e := range d {
+		e := processArray(e)
+		itemType := fmt.Sprint(reflect.TypeOf(e.Value))
+		itemStr := fmt.Sprint(e.Value)
+		if strings.Contains(itemStr, "$each") {
```

Comment:
string operation is expensive. I remember we have some utils function in the repo to check bson element existing <- we can probably leverage that

Reply:
fixed, also fixed the watcher